### PR TITLE
merge dev/cdk onto terraform

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** merge dev/cdk onto main. All previous changes have been PR reviewed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by vasireddy99 — APPROVED** [review]

File: cdk_infra/lib/stacks/eks/ec2-cluster-stack.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    this.cluster = new eks.Cluster(this, props.name, {
+      clusterName: props.name,
+      vpc: props.vpc,
+      vpcSubnets: [{ subnetType: ec2.SubnetType.PUBLIC }],
+      defaultCapacity: 0, // we want to manage capacity ourselves
+      version: props.version,
+      clusterLogging: logging,
+      kubectlLayer: GetLayer(this, props.version)
+    });
+    const clusterNodeGroup = new Nodegroup(this, 'managed-ng', {
```
Comment (Aneurysm9) [inline cdk_infra/lib/stacks/eks/ec2-cluster-stack.ts]:
```suggestion
    const clusterNodeGroup = new Nodegroup(this, '{props.name}-managed-ng', {
```

The MNG name should include the cluster name.

File: cdk_infra/lib/stacks/eks/ec2-cluster-stack.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      eks.ClusterLoggingTypes.AUDIT,
+      eks.ClusterLoggingTypes.AUTHENTICATOR,
+      eks.ClusterLoggingTypes.CONTROLLER_MANAGER,
+      eks.ClusterLoggingTypes.SCHEDULER
+    ];
+    const instance_type = props.instance_type.toLowerCase();
+    this.cluster = new eks.Cluster(this, props.name, {
+      clusterName: props.name,
+      vpc: props.vpc,
+      vpcSubnets: [{ subnetType: ec2.SubnetType.PUBLIC }],
```
Comment (Aneurysm9) [inline cdk_infra/lib/stacks/eks/ec2-cluster-stack.ts:25]: Do these need to be public?

File: cdk_infra/lib/stacks/vpc/vpc-stack.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  vpc: ec2.Vpc;
+
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+    this.vpc = new ec2.Vpc(this, 'EKSVpc', {
+      cidr: '10.0.0.0/16',
+      natGateways: 1,
+      vpnGateway: true,
+      //https://github.com/aws/aws-cdk/issues/21690
+      availabilityZones: Stack.of(this).availabilityZones.sort().slice(0, 1),
```
Comment (Aneurysm9) [inline cdk_infra/lib/stacks/vpc/vpc-stack.ts:14]: Does this only create subnets in one AZ?  We should have at least three separate AZs.

Comment (bryan-aguilar) [reply cdk_infra/lib/stacks/vpc/vpc-stack.ts:14]: No, it creates it in multiple AZs but is determined using runtime context. See the file `cdk.context.json` for AZs used at deploy time.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**